src/biokbase/narrative/jobs/jobmanager.py

```diff
@@ -516,30 +780,62 @@ def retry_jobs(self, job_id_list: List[str]) -> dict:
         retry_states = self._construct_job_output_state_set(
             retry_ids, self._create_jobs(retry_ids)  # add to self._running_jobs index
         )
-        job_states = {**orig_states, **retry_states}
 
         results_by_job_id = {}
         # fill in the job state details
         for result in retry_results:
             job_id = result["job_id"]
-            results_by_job_id[job_id] = {"job_id": job_id, "job": job_states[job_id]}
+            results_by_job_id[job_id] = {"job_id": job_id, "job": orig_states[job_id]}
             if "retry_id" in result:
                 retry_id = result["retry_id"]
-                results_by_job_id[job_id]["retry_id"] = retry_id
-                results_by_job_id[job_id]["retry"] = job_states[retry_id]
+                results_by_job_id[job_id].update(
+                    {"retry_id": retry_id, "retry": retry_states[retry_id]}
+                )
             if "error" in result:
                 results_by_job_id[job_id]["error"] = result["error"]
-        return self.add_errors_to_results(results_by_job_id, error_ids)
+        return self.add_errors_to_results(results_by_job_id, error_ids, OutputStateErrMsg.NOT_FOUND)
 
-    def add_errors_to_results(self, results: dict, error_ids: List[str]) -> dict:
-        """
-        Add the generic "not found" error for each job_id in error_ids
+    @staticmethod
+    def add_errors_to_results(
+        results: dict, error_ids: List[str], error_enum: OutputStateErrMsg, *extra_its: Tuple
+    ) -> dict:
```

Comment:
Takes the results dict, error_ids, the error enum, and any extra err msg formatting args